views/js/pciCreator/dev/audioRecordingInteraction/runtime/js/uiElements.js

```diff
@@ -102,21 +98,37 @@ define([
              * Enable the control
              */
             enable: function enable() {
-                setState(controlStates.ENABLED);
+                _setState(controlStates.ENABLED);
             },
 
             /**
              * Disable the control
              */
             disable: function disable() {
-                setState(controlStates.DISABLED);
+                _setState(controlStates.DISABLED);
             },
 
             /**
              * Activate the control
              */
             activate: function activate() {
-                setState(controlStates.ACTIVE);
+                _setState(controlStates.ACTIVE);
+            },
+
+            /**
+             * Get current state
+             * @returns {string} state - the current state
+             */
+            getState: function getState() {
+                return state;
+            },
+
+            /**
+             * Set the state of the control
+             * @param {String} newState
+             */
+            setState: function setState(newState) {
+                _setState(newState);
```

Comment:
Why not directly implement the function as a member instead of relying on a local delegation?

Reply:
Yes, i'll do it.